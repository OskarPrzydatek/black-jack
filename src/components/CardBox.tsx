import { CardColorsEnum } from '../constants/cardColors.enum';
import * as S from '../styles/CardBox.style';
import { CardColorType } from '../types/cardColor.type';
import { CardKindType } from '../types/cardKind.type';

import Icon from './Icon';

interface CardBoxProps {
  kind: CardKindType;
  color: CardColorType;
}

export default function CardBox({ kind, color }: CardBoxProps) {
  const cardKindLabel = () => {
    if (kind === 9 || kind === 10) return kind;
    return kind[0];
  };

  const cardColorStyle = () => {
    if (color === CardColorsEnum.Hearts || color === CardColorsEnum.Tiles)
      return { color: '#D91828' };
    return { color: '#2F2F2F' };
  };

  return (
    <S.CardBox>
      <S.CardLabel>
        <span data-testid="card-box-label" style={cardColorStyle()}>
          {cardKindLabel()}
        </span>
      </S.CardLabel>
      <S.CardColor>
        <Icon cardColor={color} />
      </S.CardColor>
      <S.BottomCardLabel>
        <span style={cardColorStyle()}>{cardKindLabel()}</span>
      </S.BottomCardLabel>
    </S.CardBox>
  );
}
