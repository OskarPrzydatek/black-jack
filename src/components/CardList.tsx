import React from 'react';

import * as S from '../styles/CardList.style';

interface CardListProps {
  children: React.ReactNode;
  dataTestID?: string;
}

export default function CardList({ children, dataTestID }: CardListProps) {
  return <S.CardList data-testid={dataTestID}>{children}</S.CardList>;
}
