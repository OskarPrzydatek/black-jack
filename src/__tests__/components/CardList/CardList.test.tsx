import renderer from 'react-test-renderer';
import CardBox from '../../../components/CardBox';

import CardList from '../../../components/CardList';
import { CardColorsEnum } from '../../../constants/cardColors.enum';

describe('CardList', () => {
  test('component snapshot', () => {
    const tree = renderer
      .create(
        <CardList>
          <CardBox kind={9} color={CardColorsEnum.Clovers} />
        </CardList>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

export {};
