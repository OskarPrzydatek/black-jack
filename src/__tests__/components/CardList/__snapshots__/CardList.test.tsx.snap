// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component snapshot 1`] = `
<ul
  className="sc-eDvSVe cbCSYU"
>
  <li
    className="sc-bcXHqe kFDUXW"
  >
    <div
      className="sc-gswNZR boNMTz"
    >
      <span
        data-testid="card-box-label"
        style={
          Object {
            "color": "#2F2F2F",
          }
        }
      >
        9
      </span>
    </div>
    <div
      className="sc-hLBbgP biwgZj"
    >
      <svg
        fill="#2F2F2F"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.5 12.5a3.493 3.493 0 0 1-2.684-1.254 19.92 19.92 0 0 0 1.582 2.907c.231.35-.02.847-.438.847H6.04c-.419 0-.67-.497-.438-.847a19.919 19.919 0 0 0 1.582-2.907 3.5 3.5 0 1 1-2.538-5.743 3.5 3.5 0 1 1 6.708 0A3.5 3.5 0 1 1 11.5 12.5z"
        />
      </svg>
    </div>
    <div
      className="sc-gswNZR sc-dkrFOg boNMTz kfyxUn"
    >
      <span
        style={
          Object {
            "color": "#2F2F2F",
          }
        }
      >
        9
      </span>
    </div>
  </li>
</ul>
`;
